### PR TITLE
Fixit Button Works!

### PR DESCRIPTION
**Pull request by nuwandavek**
https://github.com/user-attachments/assets/01640f80-486a-4405-bbb2-b64cf00f35ec

Todo
- [ ] Test for performance
- [x] Extension release first (handle that)

### CODE REVIEW

**Comment by ppsreejith** [comment]
Lgtm! Can you put it behind a DEV flag for now and merge it? We can remove DEV flag once extension is approved, wdyt? I'm going ahead with extension submission since there seems to be no changes in `extension` in this PR

File: web/src/package.ts [hunk 1]
```diff
@@ -2,4 +2,7 @@ export * as RPCs from './app/rpc';
 export * as utils from './helpers/utils';
 export { memoize } from './cache/cache'
 export { subscribe } from './helpers/documentSubscription';
-export { addNativeEventListener } from './helpers/nativeEvents';
\ No newline at end of file
+export { addNativeEventListener } from './helpers/nativeEvents';
+import chat from './chat/chat';
+
+export const chatModule = chat
```
Comment (ppsreejith) [inline web/src/package.ts]: Hmm, maybe check if production build works with this? `chat` pulls in a ton of dependencies. To test, you can try running: `yarn build`.

Also, just an FYI that this change cane be  one liner: `export chatModule from './chat/chat'`

**Review by ppsreejith — APPROVED** [review]
Lgtm but check if production build succeeds due to pulling in a bunch of dependencies onto `app` now

File: web/src/package.ts (hunk 1 above)
Comment (nuwandavek) [reply web/src/package.ts]: `export chatModule from './chat/chat'` 
This did not work!

It builds prod.

**Comment by nuwandavek** [comment]
@ppsreejith check this again?

File: apps/src/metabase/appState.ts [hunk 2]
```diff
@@ -10,11 +10,12 @@ import { subscribe } from "web";
 import { getRelevantTablesForSelectedDb } from "./helpers/getDatabaseSchema";
 import { querySelectorMap } from "./helpers/querySelectorMap";
 
+
 export class MetabaseState extends DefaultAppState<MetabaseAppState> {
   initialInternalState = metabaseInternalState;
   actionController = new MetabaseController(this);
 
-  public async setup() {
+  public async setup(isDev: boolean) {
```
Comment (ppsreejith) [inline apps/src/metabase/appState.ts]: I think we should just expose configs to App. I can imagine there will be configs in the future that App needs to configure + there's no dependencies on configs so we should just expose it to app from web IMO.

File: web/src/package.ts (hunk 1 above)
Comment (ppsreejith) [reply web/src/package.ts]: Fair. Also just realised we expose `getUserConfirmation` from RPCs that also has a dependency on Redux so this change doesn't change anything dependency wise Maybe we should expose ChatModule through RPCs as well (for now), wdyt?
i.e: Something like `RPCs.addUserMessage`

Maybe later we can group them and expose them as `webInterface.ts` containing: `addUserMessage` and `getUserConfirmation`. i.e: A small interface for apps to talk to the main `web` app and get a response.

**Review by ppsreejith — CHANGES_REQUESTED** [review]
Left some comments. TL;DR: We should expose configs to app & avoid adding args to fns unless necessary IMO. Plus maybe expose `addUserMessage` like we exposed `getUserConfirmation` using RPCs. So that we can later group them in `webInterface.ts` so it doesn't get out of hand in the future.

**Pull request merged**